src/client/gui/ffi/dart_ffi.cpp

```diff
@@ -214,4 +215,23 @@ catch (...)
     mpl::log(mpl::Level::warning, category, "failed converting memory to bytes");
     return -1;
 }
+
+const char* default_mount_target(const char* source)
+try
+{
+    const QString q_source{source};
+    free((void*)source);
```

Comment:
I am a little surprised that this takes ownership of the parameter and deletes memory that is supposed to hold const chars. I suppose the intention is that the client doesn't have to free after each function call, but in that case, don't you think we should remove the `const`? As it stands, it is easy to get into UB (e.g. call with a compile-time constant).

I realize this pattern is repeated all over this file, but it should be straightforward to modify everywhere (provided it works).

Reply:
That is a very good point, thanks. That is indeed a critical mistake. I'll fix it everywhere ASAP. I'll make a new PR for fixing those issues and set that PR as the base for this one. Is that ok?

Reply:
Sure, that sounds perfectly fine :smiley: